File: IAIso-v5.0/core/iaiso-java/iaiso-coordination/src/main/java/io/iaiso/coordination/Snapshot.java
package io.iaiso.coordination;

import java.util.Collections;
import java.util.Map;
import java.util.TreeMap;

/** Read-only view of coordinator state. */
public final class Snapshot {
    private final String coordinatorId;
    private final double aggregatePressure;
    private final CoordinatorLifecycle lifecycle;
    private final int activeExecutions;
    private final Map<String, Double> perExecution;

    public Snapshot(String coordinatorId, double aggregatePressure,
                    CoordinatorLifecycle lifecycle, int activeExecutions,
                    Map<String, Double> perExecution) {
        this.coordinatorId = coordinatorId;
        this.aggregatePressure = aggregatePressure;
        this.lifecycle = lifecycle;
        this.activeExecutions = activeExecutions;
        this.perExecution = Collections.unmodifiableMap(new TreeMap<>(perExecution));
    }

    public String getCoordinatorId() { return coordinatorId; }
    public double getAggregatePressure() { return aggregatePressure; }
    public CoordinatorLifecycle getLifecycle() { return lifecycle; }
    public int getActiveExecutions() { return activeExecutions; }
    public Map<String, Double> getPerExecution() { return perExecution; }
}

File: IAIso-v5.0/core/iaiso-csharp/src/Iaiso.Coordination/Snapshot.cs
using System.Collections.Generic;

namespace Iaiso.Coordination;

/// <summary>Read-only view of coordinator state.</summary>
public sealed class Snapshot
{
    public string CoordinatorId { get; }
    public double AggregatePressure { get; }
    public CoordinatorLifecycle Lifecycle { get; }
    public int ActiveExecutions { get; }
    public IReadOnlyDictionary<string, double> PerExecution { get; }

    public Snapshot(string coordinatorId, double aggregatePressure,
                    CoordinatorLifecycle lifecycle, int activeExecutions,
                    IReadOnlyDictionary<string, double> perExecution)
    {
        CoordinatorId = coordinatorId;
        AggregatePressure = aggregatePressure;
        Lifecycle = lifecycle;
        ActiveExecutions = activeExecutions;
        PerExecution = new SortedDictionary<string, double>(
            new Dictionary<string, double>(perExecution));
    }
}
